src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -217,6 +263,216 @@ await contextualFilter.EvaluateAsync(context, appContext).ConfigureAwait(false)
+        }
+
+        private async ValueTask<Variant> GetVariantAsync(string feature, TargetingContext context, bool useContext, CancellationToken cancellationToken)
+        {
+            FeatureDefinition featureDefinition = await _featureDefinitionProvider
+                .GetFeatureDefinitionAsync(feature)
+                .ConfigureAwait(false);
+
+            if (featureDefinition == null)
+            {
+                throw new FeatureManagementException(
+                    FeatureManagementError.MissingFeature,
+                    $"The feature declaration for the feature '{feature}' was not found.");
+            }
+
+            if (featureDefinition.Allocation == null)
+            {
+                throw new FeatureManagementException(
+                    FeatureManagementError.MissingAllocation,
+                    $"No allocation is defined for the feature {featureDefinition.Name}");
+            }
+
+            if (!featureDefinition.Variants?.Any() ?? false)
+            {
+                throw new FeatureManagementException(
+                    FeatureManagementError.MissingVariantDefinition,
+                    $"No variants are registered for the feature {feature}");
+            }
+
+            VariantDefinition variantDefinition;
+
+            bool isFeatureEnabled = await IsEnabledAsync(feature, context, true, cancellationToken).ConfigureAwait(false);
+
+            variantDefinition = await GetVariantDefinitionAsync(featureDefinition, context, useContext, isFeatureEnabled, cancellationToken).ConfigureAwait(false);
+
+            if (variantDefinition == null)
+            {
+                return null;
+            }
+
+            IConfigurationSection variantConfiguration = null;
+
+            bool configValueSet = !string.IsNullOrEmpty(variantDefinition.ConfigurationValue);
+            bool configReferenceValueSet = !string.IsNullOrEmpty(variantDefinition.ConfigurationReference);
+
+            if (configValueSet && configReferenceValueSet)
+            {
+                throw new FeatureManagementException(
+                    FeatureManagementError.InvalidVariantConfiguration,
+                    $"Both ConfigurationValue and ConfigurationReference are specified for the variant {variantDefinition.Name} in feature {feature}");
+            }
+            else if (configReferenceValueSet)
+            {
+                if (_configuration == null)
+                {
+                    throw new FeatureManagementException(
+                        FeatureManagementError.InvalidVariantConfiguration,
+                        $"Cannot use {nameof(variantDefinition.ConfigurationReference)} if no instance of {nameof(IConfiguration)} is present.");
+                }
+
+                variantConfiguration = _configuration.GetSection(variantDefinition.ConfigurationReference);
+            }
+            else if (configValueSet)
+            {
```

Comment:
What's the point of passing the empty string?

Reply:
Related to [this](https://github.com/microsoft/FeatureManagement-Dotnet/pull/250#discussion_r1295021301) since I wanted to follow the existing pattern

Reply:
seems like public path constructor isn't needed